hamilton/cli/__main__.py

```diff
@@ -0,0 +1,188 @@
+import logging
+from pathlib import Path
+from pprint import pprint
+from typing import Annotated, Any, Optional
+
+import typer
+from pydantic import BaseModel
+
+from hamilton import driver
+from hamilton.cli import commands
+
+logger = logging.getLogger(__name__)
+
+
+class Response(BaseModel):
+    command: str
+    success: bool
+    message: Any
+
+
+class CliState:
+    verbose: Optional[bool] = None
+    json_out: Optional[bool] = (None,)
+    dr: Optional[driver.Driver] = None
+    dataflow_version: Optional[dict] = None
+
+
+cli = typer.Typer()
+state = CliState()
+
+
+# entrypoint for `hamilton` without command
+@cli.callback()
```

Comment:
Logging statements of what's happening might be nice -- E.G> `attempting to build`, `attempting to download prior git revision`, etc..